vgsshow/src/main/java/com/verygoodsecurity/vgsshow/widget/view/pdf/VGSPDFView.kt

```diff
@@ -0,0 +1,206 @@
+
+    /** Enable/Disable double tap. */
+    var isDoubleTapEnabled: Boolean = DOUBLE_TAB_ENABLED
+
+    /** Improve rendering on low-res screens. */
+    var isAntialiasingEnabled: Boolean = ANTIALIAS_ENABLED
+
+    /** Spacing between pages in dp. */
+    var spacing: Int = SPACING
+
+    /** Register a callback to be invoked when documents is loaded successfully. */
+    var onLoadCompleteListener: OnLoadCompleteListener? = null
+
+    /** Register a callback to be invoked when documents loading failed. */
+    var onErrorListener: OnErrorListener? = null
+
+    /** Return true if document was revealed. */
+    var hasDocument: Boolean = false
+        private set
+
+    init {
+
+        context.getStyledAttributes(attrs, R.styleable.VGSPDFView) {
+            defaultPage = getInt(R.styleable.VGSPDFView_defaultPage, DEFAULT_PAGE)
+            isSwipeEnabled = getBoolean(R.styleable.VGSPDFView_isSwipeEnabled, SWIPE_ENABLED)
+            isSwipeHorizontalEnabled = getBoolean(
+                R.styleable.VGSPDFView_isSwipeHorizontalEnabled,
+                SWIPE_HORIZONTAL_ENABLED
+            )
+            isDoubleTapEnabled =
+                getBoolean(R.styleable.VGSPDFView_isDoubleTapEnabled, DOUBLE_TAB_ENABLED)
+            isAntialiasingEnabled =
+                getBoolean(R.styleable.VGSPDFView_isAntialiasingEnabled, ANTIALIAS_ENABLED)
+            spacing = getInt(R.styleable.VGSPDFView_spacing, 0)
+        }
+    }
+
+    override fun getFieldType() = VGSFieldType.PDF
+
+    override fun createChildView() = PDFView(context, null)
+
+    override fun saveState(state: Parcelable?) = VGSPDFViewState(state).apply {
+        this.hasDocument = this@VGSPDFView.hasDocument
+        this.defaultPage = this@VGSPDFView.defaultPage
+        this.isSwipeEnabled = this@VGSPDFView.isSwipeEnabled
+        this.isSwipeHorizontalEnabled = this@VGSPDFView.isSwipeHorizontalEnabled
+        this.isDoubleTapEnabled = this@VGSPDFView.isDoubleTapEnabled
+        this.isAntialiasingEnabled = this@VGSPDFView.isAntialiasingEnabled
+        this.isAntialiasingEnabled = this@VGSPDFView.isAntialiasingEnabled
+        this.spacing = this@VGSPDFView.spacing
+    }
+
+    override fun restoreState(state: BaseSavedState) {
+        (state as? VGSPDFViewState)?.let {
+            this@VGSPDFView.hasDocument = it.hasDocument
+            this@VGSPDFView.defaultPage = it.defaultPage
+            this@VGSPDFView.isSwipeEnabled = it.isSwipeEnabled
+            this@VGSPDFView.isSwipeHorizontalEnabled = it.isSwipeHorizontalEnabled
+            this@VGSPDFView.isDoubleTapEnabled = it.isDoubleTapEnabled
+            this@VGSPDFView.isAntialiasingEnabled = it.isAntialiasingEnabled
+            this@VGSPDFView.isAntialiasingEnabled = it.isAntialiasingEnabled
+            this@VGSPDFView.spacing = it.spacing
+            refresh()
+        }
+    }
+
+    /**
+     * Should be called if document already rendered and any changes in setting are made.
```

Comment:
Call this method when some configuration has been made after rendering PDF content. View will rerender content according to a new setup.